tests/decorators/test_prompt_decorator.py

```diff
@@ -0,0 +1,259 @@
+        # This should be documented somewhere or changed
+        replicate.default_client._api_token = os.getenv("REPLICATE_API_KEY")
+        try:
+            output = ""
+            for event in replicate.run(
+                model,
+                input={
+                    "prompt": messages[0]["content"] + " " + messages[-1]["content"],
+                    "temperature": 0.8,
+                },
+            ):
+                output += str(event)
+        except ReplicateModelError:
+            pytest.skip("Replicate not available")
+        if not output:
+            pytest.skip("Replicate not available")
+        return output
+    raise ValueError(f"Unknown provider: {provider}")
+
+
+# NOTE: prompt is a decorator, but for brevity, it's used as a higher-order function in tests
+_call_llm = prompt(
+    path=None,
+    template="You are an assistant on the following topics: {topics}.",
+)(_call_llm_base)
+_call_llm_with_defaults = prompt(
+    path=None,
+    template="You are an assistant on the following topics: {topics}.",
+    temperature=0.9,
+    top_p=0.1,
+)(_call_llm_base)
+
+
+@pytest.mark.parametrize("provider_model", _PROVIDER_AND_MODEL)
+def test_prompt_decorator(
+    provider_model: tuple[str, str],
+    opentelemetry_test_configuration: tuple[Tracer, InMemorySpanExporter],
+    call_llm_messages: list[ChatCompletionMessageParam],
+):
+    provider, model = provider_model
+    # GIVEN an OpenTelemetry configuration without HumanloopSpanProcessor
+    _, exporter = opentelemetry_test_configuration
+    # WHEN using the Prompt decorator
+    _call_llm(
+        provider=provider,
+        model=model,
+        messages=call_llm_messages,
+    )
+    # THEN two spans are created: one for the OpenAI LLM provider call and one for the Prompt
+    spans = exporter.get_finished_spans()
+    assert len(spans) == 2
+    assert not is_humanloop_span(span=spans[0])
+    assert is_humanloop_span(span=spans[1])
+    # THEN the Prompt span is not enhanced with information from the LLM provider
+    assert is_humanloop_span(spans[1])
+    # THEN no information is added to the Prompt span without the HumanloopSpanProcessor
+    assert spans[1].attributes.get("prompt") is None  # type: ignore
+
+
+@pytest.mark.parametrize("provider_model", _PROVIDER_AND_MODEL)
+def test_prompt_decorator_with_hl_processor(
+    provider_model: tuple[str, str],
+    opentelemetry_hl_test_configuration: tuple[Tracer, InMemorySpanExporter],
+    call_llm_messages: list[ChatCompletionMessageParam],
+):
+    provider, model = provider_model
+    # GIVEN an OpenTelemetry configuration with HumanloopSpanProcessor
+    _, exporter = opentelemetry_hl_test_configuration
+    # WHEN using the Prompt decorator
+    _call_llm(
+        provider=provider,
+        model=model,
+        messages=call_llm_messages,
+    )
+    # THEN two spans are created: one for the OpenAI LLM provider call and one for the Prompt
```

Comment:
Another point here: the most dogmatic approach would be to filter out Spans from the `HLProcessor,` so they never reach the `HLExporter.` On the other hand, if we want to make this feature capable of integrating with a customer's existing OTel telemetry solution, it would be rude to drop Spans without their knowledge.

Seeking opinions on this

Reply:
Resolving this @andreibratu please but please confirm you have added to linear